### ISSUE TITLE
failed download leads to checksum error

### ISSUE THREAD

**Issue by till** [hashicorp/packer#146]
It seems like, I've been having some issues download an ISO file from an ubuntu mirror and it took me a while to get this: whenever the download fails, packer will exit with an error that the checksum didn't match. 

Kind of true (because the broken download's checksum is indeed different), but also misleading about what happened.

Been downloading the ISO with wget and noticed how it constantly has to resume. I also saw in another issue that the `iso_url` can be a file URL — maybe add that to the documentation?

Here the output:

```
virtualbox: Download progress: 24%
virtualbox: Download progress: 24%
virtualbox: Download progress: 24%
virtualbox: Download progress: 24%
virtualbox: Download progress: 24%
==> virtualbox: Error downloading ISO: checksums didn't match expected: 9b218654cdcdf9722171648c52f8a088
Build 'virtualbox' errored: Error downloading ISO: checksums didn't match expected: 9b218654cdcdf9722171648c52f8a088

==> Some builds didn't complete successfully and had errors:
--> virtualbox: Error downloading ISO: checksums didn't match expected: 9b218654cdcdf9722171648c52f8a088

==> Builds finished but no artifacts were created.
```

Here is the relevant piece of the same command prefixed with `PACKER_LOG=1`:

```
2013/07/06 19:55:17 ui: virtualbox: Download progress: 19%
virtualbox: Download progress: 19%
2013/07/06 19:55:21 /Users/till/packer/packer-builder-virtualbox: 2013/07/06 19:55:21 Verifying checksum of /Users/till/Documents/workspaces//dev-vm/packer_cache/0c02b0d831fe9746a650c270da0a5ee96dcc4ed89b33f4c3e7b6a19856a0265c
2013/07/06 19:55:21 ui error: ==> virtualbox: Error downloading ISO: checksums didn't match expected: 9b218654cdcdf9722171648c52f8a088
==> virtualbox: Error downloading ISO: checksums didn't match expected: 9b218654cdcdf9722171648c52f8a088
2013/07/06 19:55:21 ui error: Build 'virtualbox' errored: Error downloading ISO: checksums didn't match expected: 9b218654cdcdf9722171648c52f8a088
Build 'virtualbox' errored: Error downloading ISO: checksums didn't match expected: 9b218654cdcdf9722171648c52f8a088
2013/07/06 19:55:21 /Users/till/packer/packer-command-build: 2013/07/06 19:55:21 Builds completed. Waiting on interrupt barrier...
```
